velox/dwio/common/CachedBufferedInput.h

```diff
@@ -200,9 +195,8 @@ class CachedBufferedInput : public BufferedInput {
   std::vector<std::shared_ptr<cache::CoalescedLoad>> allCoalescedLoads_;
 
   const uint64_t fileSize_;
-  const int32_t loadQuantum_;
-  const int32_t maxCoalesceDistance_;
   int64_t prefetchSize_{0};
+  const std::shared_ptr<ReaderOptions> options_;
```

Comment:
Is there a reason this is a shared_ptr rather than just a raw struct reference?  A shared_ptr seems more dangerous as external changes could change this property set.

Reply:
Generally I agree, however...in this case I am working with what I have.  Passing a storing a const reference to a actual object is not (in my opinion) better than a shared pointer for future expansion. Given the current state of things I would prefer to pass a shared_ptr instead of an const object reference. I think maybe at some point to resolve all of the "config" issues we should rewrite the way configuration is handled across Velox in a common uniformed way.